phenol-annotations/src/main/java/org/monarchinitiative/phenol/annotations/formats/hpo/DiseaseToGeneAssociation.java

```diff
@@ -16,38 +17,54 @@
   /**
    * The CURIE representing a disease, e.g., OMIM:600100.
    */
-  private final TermId disease;
+  private final TermId diseaseId;
   private final List<GeneToAssociation> gene2assoc;
 
-  public DiseaseToGeneAssociation(TermId id, List<GeneToAssociation> genes) {
-    this.gene2assoc = genes;
-    this.disease = id;
+  public static DiseaseToGeneAssociation of(TermId diseaseId, List<GeneToAssociation> geneToAssociations) {
+    return new DiseaseToGeneAssociation(diseaseId, geneToAssociations);
   }
 
-  public TermId getDiseaseId() {
-    return disease;
+  private DiseaseToGeneAssociation(TermId diseaseId, List<GeneToAssociation> geneAssociations) {
+    this.diseaseId = Objects.requireNonNull(diseaseId, "Disease id must not be null");
+    this.gene2assoc = Objects.requireNonNull(geneAssociations, "Gene associations must not be null");
+  }
+
+  public TermId diseaseId() {
+    return diseaseId;
+  }
+
+  public List<GeneToAssociation> associations() {
+    return gene2assoc;
   }
 
   /**
-   * @return a list of all genes (regardless of bla32 type).
+   * @return a list of all genes (regardless of the association type).
    */
-  public List<Gene> getGeneList() {
-    return gene2assoc.stream().map(GeneToAssociation::getGene).collect(Collectors.toList());
+  public List<GeneIdentifier> getGeneList() {
```

Comment:
Given you've broken the API already, might as well update the method signature to `public List<GeneIdentifier> getGeneIdentifiers()`